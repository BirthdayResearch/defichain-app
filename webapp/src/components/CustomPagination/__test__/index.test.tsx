import React from 'react';
import CustomPaginationComponent from '../index';
import { shallow } from 'enzyme';

describe('CustomPaginationComponent component', () => {
  it('should check for snapshot', () => {
    const props = {
      data: [],
      currentPage: 1,
      pagesCount: 1,
      label: 'label',
      handlePageClick: () => {},
      showNextOnly: true,
      disableNext: false,
      cancelToken: 1,
    };

    const wrapper = shallow(
      <CustomPaginationComponent
        data={props.data}
        currentPage={props.currentPage}
        pagesCount={props.pagesCount}
        label={props.label}
        handlePageClick={props.handlePageClick}
        showNextOnly={props.showNextOnly}
        disableNext={props.disableNext}
        cancelToken={props.cancelToken}
      />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
