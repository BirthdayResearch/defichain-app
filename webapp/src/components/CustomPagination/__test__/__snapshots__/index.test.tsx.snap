// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CustomPaginationComponent component should check for snapshot 1`] = `
<div
  className="d-flex text-right justify-content-between align-items-center mt-3"
>
  <div>
    label
  </div>
  <Pagination
    aria-label="pagination"
    listTag="ul"
    tag="nav"
  >
    <PaginationItem
      disabled={true}
      tag="li"
    >
      <PaginationLink
        onClick={[Function]}
        previous={true}
        tag="a"
      >
        <MdChevronLeft />
      </PaginationLink>
    </PaginationItem>
    <PaginationItem
      active={true}
      key="1"
      tag="li"
    >
      <PaginationLink
        onClick={[Function]}
        tag="a"
      >
        1
      </PaginationLink>
    </PaginationItem>
    <PaginationItem
      disabled={true}
      tag="li"
    >
      <PaginationLink
        disabled={false}
        next={true}
        onClick={[Function]}
        tag="a"
      >
        <MdChevronRight />
      </PaginationLink>
    </PaginationItem>
  </Pagination>
</div>
`;
